### DISCUSSION TITLE
Can I connect a dialogic character to my own character class?

### DISCUSSION THREAD

**Discussion by scoulston** [dialogic-godot/dialogic#1913] [Q&A]
Hello! I am working on a sort of text based adventure game (but not purely text), and I have a character data class I'm using to store character data on the characters for my game:

`extends Resource
class_name CharacterData

#starting inventory
const DEFAULT_INV = preload("res://inventory/default_inventory.tres")
#starting skills
const DEFAULT_SKILLS = preload("res://character/skill/character_skills/default_character_skills.tres")

## the name of the character
@export var name: String = ""
## the inventory for the character
@export var inv: InventoryData = DEFAULT_INV
## list of skills and their levels/values for the character (besides constitution?)
@export var skills: CharacterSkillsData = DEFAULT_SKILLS
## dialog color for the character
@export var color: Color
## health of the character
@export var current_hp: int
## maximum health of the character
@export var max_hp: int = 1`

I've just starting working with Godot, and I was wondering if you guys had any ideas as to how I can connect my character data to dialogic character data? The idea I guess is to be able to use my character data ( hp, max hp, the inventory of items the character has, etc) within dialogic. If that is possible. I'm not even sure at this point if that makes sense, but I figured I would ask for input. Thanks in advance!

**Comment by CakeVR** [comment]
Hello, first, I would recommend to put your code in three backticks.
You can also mark it via the `<>` button, same result. After the initial three backticks, you can write `gdscript`, to achieve syntax highlighting.
![image](https://github.com/coppolaemilio/dialogic/assets/65981767/ea88f61f-2c7e-4eaa-ab35-b13fe89d8b63)

About your question, I recommend to keep the data of Dialogic and your game logic separate but use the same character name as keys.

**Example**: You store your character data *somehow* (a `Dictionary`?) and then access the correct data with the character name as `key`.
Then, you can make these changes happen with the `Call event`:
![image](https://github.com/coppolaemilio/dialogic/assets/65981767/6679633a-7142-4ff3-be6d-ce7cfbf56c72)

If you need more help, let me know and provide some specifics on how you imagine to change and alter your stats.
